Error compiling adi_zynqmp_defconfig for arm64

Hello, 

i'm trying to compile adi_zynqmp_defconfig for arm64 but i get the next error log: 

i have been viewing the code and i have seen that the struct that fails (iio_buffer) was edited recently. 

  CC      drivers/misc/jesd204b/jesd_phy.o
  CC      drivers/misc/jesd204b/gtx7s_cpll_bands.o
  CC      drivers/misc/jesd204b/gtx7s_qpll_bands.o
  AR      drivers/misc/jesd204b/jesd204b_phy.o
  CC      drivers/misc/jesd204b/xilinx_jesd204b.o
  AR      drivers/misc/jesd204b/jesd204b.o
  AR      drivers/misc/jesd204b/built-in.o
  AR      drivers/misc/lis3lv02d/built-in.o
  AR      drivers/misc/mathworks/built-in.o
  CC [M]  drivers/misc/mathworks/mathworks_ip_common.o
  CC [M]  drivers/misc/mathworks/mathworks_generic_of.o
  CC [M]  drivers/misc/mathworks/mathworks_generic_pci.o
  CC [M]  drivers/misc/mathworks/mathworks_ipcore.o
  LD [M]  drivers/misc/mathworks/mwipcore.o
  CC [M]  drivers/misc/mathworks/mw_stream_channel.o
In file included from ./include/linux/dmaengine.h:20:0,
                 from drivers/misc/mathworks/mw_stream_channel.h:12,
                 from drivers/misc/mathworks/mw_stream_channel.c:12:
drivers/misc/mathworks/mw_stream_channel.c: In function ‘mwadma_allocate_desc’:
drivers/misc/mathworks/mw_stream_channel.c:90:26: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 5 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
     dev_dbg(&mwchan->dev,"buf_phys_addr 0x%08lx, size %u\n", (unsigned long) tmp->phys, tmp->length);
                          ^                                                              ~~~~~~
./include/linux/device.h:1360:31: note: in definition of macro ‘dev_dbg’
   dev_printk(KERN_DEBUG, dev, format, ##arg); \
                               ^~~~~~
drivers/misc/mathworks/mw_stream_channel.c: In function ‘mwadma_start’:
drivers/misc/mathworks/mw_stream_channel.c:296:86: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 4 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
         dev_err(&mwchan->dev,"prep_slave_single failed: buf_phys_addr 0x%08lx, size %u\n", (unsigned long) mwchan->curr->phys, mwchan->curr->length);
                                                                                     ~^                                         ~~~~~~~~~~~~~~~~~~~~
                                                                                     %lu
drivers/misc/mathworks/mw_stream_channel.c: In function ‘mw_stream_chan_show’:
drivers/misc/mathworks/mw_stream_channel.c:1050:29: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 3 has type ‘long int’ [-Wformat=]
     return sprintf(buf, "%llu\n", atomic64_read(&rxcount));
                          ~~~^
                          %lu
  LD [M]  drivers/misc/mathworks/mwipcore_dma_streaming.o
  CC [M]  drivers/misc/mathworks/mw_stream_iio_channel.o
In file included from drivers/misc/mathworks/mw_stream_iio_channel.c:12:0:
./include/linux/iio/buffer-dma.h:51:26: error: field ‘block’ has incomplete type
  struct iio_buffer_block block;
                          ^~~~~
./include/linux/iio/buffer-dma.h:101:20: error: field ‘buffer’ has incomplete type
  struct iio_buffer buffer;
                    ^~~~~~
./include/linux/iio/buffer-dma.h:156:9: warning: ‘struct iio_buffer_block_alloc_req’ declared inside parameter list will not be visible outside of this definition or declaration
  struct iio_buffer_block_alloc_req *req);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/misc/mathworks/mw_stream_iio_channel.c: In function ‘mw_stream_iio_buffer_preenable’:
drivers/misc/mathworks/mw_stream_iio_channel.c:114:86: error: dereferencing pointer to incomplete type ‘struct iio_buffer’
   mw_ip_write32(mwchan->mwdev->mw_ip_info, mwchan->tlast_cntr_addr, indio_dev->buffer->length);
                                                                                      ^~
scripts/Makefile.build:314: fallo en las instrucciones para el objetivo 'drivers/misc/mathworks/mw_stream_iio_channel.o'
make[3]: *** [drivers/misc/mathworks/mw_stream_iio_channel.o] Error 1
scripts/Makefile.build:573: fallo en las instrucciones para el objetivo 'drivers/misc/mathworks'
make[2]: *** [drivers/misc/mathworks] Error 2
scripts/Makefile.build:573: fallo en las instrucciones para el objetivo 'drivers/misc'
make[1]: *** [drivers/misc] Error 2
Makefile:1024: fallo en las instrucciones para el objetivo 'drivers'
make: *** [drivers] Error 2

@commodo @larsclausen

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,6 +18,8 @@ env:
     DTS_FILES=arch/arm/boot/dts/zynq-*.dts IMAGE=uImage
   - DEFCONFIG_NAME=adi_zynqmp_defconfig ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu-
     DTS_FILES=arch/arm64/boot/dts/xilinx/zynqmp-*.dts DTS_PREFIX=xilinx/ IMAGE=Image
+  - COMPILE_TEST=y DEFCONFIG_NAME=zynq_xcomm_adv7511_defconfig ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-
+  - COMPILE_TEST=y DEFCONFIG_NAME=adi_zynqmp_defconfig ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu-
 
 before_install:
   - sudo apt-get update -qq

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,9 +24,4 @@ before_install:
   - sudo apt-get install -y build-essential bc u-boot-tools gcc-arm-linux-gnueabihf gcc-aarch64-linux-gnu
 
 script:
-  - if [[ -n "$TRAVIS_BRANCH" ]]; then git fetch origin +refs/heads/${TRAVIS_BRANCH}:${TRAVIS_BRANCH} ; fi
-  - COMMIT_RANGE=$([ "$TRAVIS_PULL_REQUEST" == "false" ] &&  echo HEAD || echo ${TRAVIS_BRANCH}..)
-  - make ${DEFCONFIG_NAME}
-  - make -j`getconf _NPROCESSORS_ONLN` $IMAGE UIMAGE_LOADADDR=0x8000
-  - for file in $DTS_FILES; do make ${DTS_PREFIX}`basename $file | sed  -e 's\dts\dtb\g'` || exit 1;done
-  - scripts/checkpatch.pl --git ${COMMIT_RANGE} --ignore FILE_PATH_CHANGES --ignore LONG_LINE --ignore LONG_LINE_STRING --ignore LONG_LINE_COMMENT
+  - ./ci/travis/run-build.sh

--- a/ci/travis/run-build.sh
+++ b/ci/travis/run-build.sh
@@ -22,6 +22,15 @@ build_default() {
 		--ignore LONG_LINE_COMMENT
 }
 
+build_compile_test() {
+	make ${DEFCONFIG_NAME}
+	make -j`getconf _NPROCESSORS_ONLN`
+}
+
 BUILD_TYPE=default
+
+if [ "$COMPILE_TEST" == "y" ] ; then
+	BUILD_TYPE=compile_test
+fi
 
 build_${BUILD_TYPE}

--- a/ci/travis/run-build.sh
+++ b/ci/travis/run-build.sh
@@ -1,0 +1,27 @@
+#!/bin/bash
+set -xe
+
+build_default() {
+	if [ -n "$TRAVIS_BRANCH" ]; then
+		git fetch origin +refs/heads/${TRAVIS_BRANCH}:${TRAVIS_BRANCH}
+	fi
+
+	COMMIT_RANGE=$([ "$TRAVIS_PULL_REQUEST" == "false" ] &&  echo HEAD || echo ${TRAVIS_BRANCH}..)
+
+	make ${DEFCONFIG_NAME}
+	make -j`getconf _NPROCESSORS_ONLN` $IMAGE UIMAGE_LOADADDR=0x8000
+
+	for file in $DTS_FILES; do
+		make ${DTS_PREFIX}`basename $file | sed  -e 's\dts\dtb\g'` || exit 1
+	done
+
+	scripts/checkpatch.pl --git ${COMMIT_RANGE} \
+		--ignore FILE_PATH_CHANGES \
+		--ignore LONG_LINE \
+		--ignore LONG_LINE_STRING \
+		--ignore LONG_LINE_COMMENT
+}
+
+BUILD_TYPE=default
+
+build_${BUILD_TYPE}

--- a/drivers/misc/mathworks/mw_stream_iio_channel.c
+++ b/drivers/misc/mathworks/mw_stream_iio_channel.c
@@ -8,7 +8,7 @@
 
 #include <linux/iio/iio.h>
 #include <linux/iio/sysfs.h>
-#include <linux/iio/buffer.h>
+#include <linux/iio/buffer_impl.h>
 #include <linux/iio/buffer-dma.h>
 #include <linux/iio/buffer-dmaengine.h>
 #include <linux/idr.h>

